gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -1,52 +1,83 @@
-        }
-
-        try (RestHighLevelClient client = new RestHighLevelClient(
-                RestClient.builder(
-                        new HttpHost(parameters.getOptionalString("primaryHostName").or("localhost"),
-                                parameters.getOptionalPositiveInteger("primaryPort").or(9200),
-                                "http"),
-                        new HttpHost(parameters.getOptionalString("secondaryHostName").or("localhost"),
-                                parameters.getOptionalPositiveInteger("secondaryPort").or(9201),
-                                "http")
-                ))) {
-            for (Map.Entry<String, String> e : sampleDocs.entrySet()) {
-                final String docId = e.getKey();
-                final String docContent = e.getValue();
-
-                // you can index documents one-by-one
-                client.index(new IndexRequest("documents", "doc", docId)
-                        .source("text", docContent));
-            }
-
-            // TODO: in practice, you will want to use BulkRequest for speed
-        }
+
+  private static final Logger log = LoggerFactory.getLogger(IndexGigawordWithElasticSearch.class);
+
+  /**
+   * @param argv takes two arguments, a path to the GZip file and an index name
+   * @throws IOException
+   */
+  public static void main(String[] argv) throws IOException {
+
+    // get arguments: gzip file path and index name
+    if (argv.length < 2) {
+      log.error("Required inputs: file path to GZip file and index name");
+      System.exit(-1);
+    }
+
+    String filePath = argv[0];
+    String indexName = argv[1];
+
+    // check filePath is valid
+    File file;
+    if (!(file = new File(filePath)).exists()) {
+      log.error("{} does not exist", filePath);
+      System.exit(-1);
+    }
+    String contentType = Files.probeContentType(file.toPath());
+    if ((contentType != null && !contentType.equalsIgnoreCase("application/gzip"))
+        || (contentType == null
+            && !FilenameUtils.getExtension(file.getName()).equalsIgnoreCase("gz"))) {
+      log.error("{} is not a valid GZip file", filePath);
+      System.exit(-1);
+    }
+
+    // start indexing
+    try {
+
+      RestHighLevelClient client =
+          new RestHighLevelClient(
+              RestClient.builder(
+                  new HttpHost("localhost", 9200, "http"),
+                  new HttpHost("localhost", 9201, "http")));
+
+      BulkRequest bulkRequest = new BulkRequest();
+      GigawordFileProcessor proc = new GigawordFileProcessor(file);
+      Iterator<Article> iterator = proc.iterator();
+      while (iterator.hasNext()) {
+        Article article = iterator.next();
+
+        bulkRequest.add(
+            new IndexRequest(indexName, "doc", article.getId()).source("text", article.getText()));
+      }
+
+      client.bulk(bulkRequest, RequestOptions.DEFAULT);
+
+      client.close();
+      client = null;
```

Comment:
You shouldn't need to do this - even if anything is to be gained from releasing the reference to `client` early, it will go out of scope at the end of the block anyway.